tiny_dnn/parameter.h

```diff
@@ -0,0 +1,125 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include "tiny_dnn/core/framework/tensor.h"
+#include "tiny_dnn/util/util.h"
+
+namespace tiny_dnn {
+
+enum class parameter_type : int8_t { weight = 0x0001, bias = 0x0002 };
+
+class parameter {
```

Comment:
Should this class carry template parameters for Tensor class instances?
Here, parameter class isn't a template class so its member Tensor variables are like this
```cpp
  Tensor<float_t> data_;
  Tensor<float_t> grad_;
```

If we look at the definition of Tensor class, its first templat parameter is `typename U = float_t` so I wonder why people explitly write `<float_t>` all over the library.

Personally, I don't want to carry template parameters all the time when I write code, so I really wish Tensor class's data type would be dynamic.